README.md

```diff
@@ -71,97 +73,304 @@ to your backend. One of tools helping with it is [graphql-cli](https://www.npmjs
+- Custom scalars are parsed as `Js.Json.t`, and can be parsed using the `@decoder` directive
 - Arguments with input objects
 - Using `@skip` and `@include` will force non-optional fields to become
   optional.
 - Unions are converted to polymorphic variants, with exhaustiveness checking.
   This only works for object types, not for unions containing interfaces.
 - Interfaces are also converted into polymorphic variants. Overlapping interface
   selections and other more uncommon use cases are not yet supported.
-- Basic fragment support
-- Required arguments validation - you're not going to miss required arguments on any field.
+- Support for fragments
+- Required arguments validation
 
-# Extra features
+## Typical use
 
-By using some directives prefixed `bs`, `graphql_ppx` lets you modify how the
-result of a query is parsed. All these directives will be removed from the query
-at compile time, so your server doesn't have to support them.
+GraphQL PPX is a utility to work with the GraphQL protocol in ReasonML.
+Typically this PPX is being used in combination with a GraphQL client. Popular
+clients include [Reason Apollo Hooks](https://github.com/Astrocoders/reason-apollo-hooks/commits/master)
+or [Reason URQL](https://github.com/FormidableLabs/reason-urql). They also
+provide a more end-to-end getting started. This documentation will focus on how
+to create queries and fragments, and parse responses.
 
-### Record conversion
+## Defining a Query
 
-While `Js.t` objects often have their advantages, they also come with some
-limitations. For example, you can't create new objects using the spread (`...`)
-syntax or pattern match on their contents. Since they are not named, they also
-result in quite large type error messages when there are mismatches.
+You can define a query in your ReasonML file with the following code
 
-Reason/OCaml records, on the other hand, can be pattern matched, created using the
-spread syntax, and give nicer error messages when they mismatch. `graphql_ppx`
-gives you the option to decode a field as a record using the `@bsRecord`
-directive:
+```reason
+[%graphql {|
+  query UserQuery {
+    user {
+      id
+      role
+    }
+  }
+|}];
+```
+
+This will create the `UserQuery` module. This module has the following
+contents assigned:
+
+### Let bindings
+
+#### Basic
+
+- `query` (`string`), the GraphQL query or mutation
+- `parse` (`UserQuery.t_raw => UserQuery.t`), the function to parse the raw
+  GraphQL response into ReasonML types.
+- `makeVariables` (`(~your, ~arguments, ()) => Js.Json.t`): a
+  function that takes labeled arguments to produce the variables that can be
+  sent together with the query. This will also validate and type-check the
+  variables.
+- `definition`: the module contents packaged. This is usually what you provide
+  to the client for ergonomics so you don't have to pass multiple arguments per
+  query
+
+#### Advanced
+
+- `serialize` (`t => t_raw`): this is the opposite of parse.
```

Comment:
This will be implemented in second PR so we have to remove that from the documentation for now.